aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/AWSS3StoragePlugin.java

```diff
@@ -82,22 +80,27 @@
      * Constructs the AWS S3 Storage Plugin initializing the executor service.
      */
     public AWSS3StoragePlugin() {
-        this(
-            (context, region, bucket) ->
-                    new AWSS3StorageService(context, region, bucket, false),
-            () -> AWSMobileClient.getInstance().getIdentityId()
-        );
+        this((context, region, bucket) ->
+                new AWSS3StorageService(context, region, bucket, new MobileClientAWSAuthProvider(), false),
+                new MobileClientAWSAuthProvider());
```

Comment:
Nit: why do both plugin and service need the reference to the auth provider? If the service is going to have access to it, then the plugin shouldn't need it right?

Also, you should keep only one reference of `MobileClientAWSAuthProvider()` rather than instantiating two separately.

Reply:
Java's requirements that `this(...)` or `super(...)` must exist in the first line --

This is one reason I use factory methods all of the the place, cause then you can do:
```
static Gizmo create() {
    allKinds();
    ofMagical();
    thangsBefore();
    return new Gizmo(kFine, iAmReady, now);
}
```

Anyway, we are stuck with a no-args constructor as our plugin instantiation API.

To "copy" with this, you could do like:
```
class GizmoPlugin {
    GizmoPlugin() {
        this(defaultFidgetSpinner(), defaultDongler());
    }

    GizmoPlugin(FidgetSpinner spinner, Dongler dongler) {
        this(spinner, dongler, spinner, dongler); // if implementing a few needed interfaces from one component
    }
  
   GizmoPlugin(
        Fidget fidget,
        AbstractDongler abstractDongler,
        Spinner spinner,
        DonglerBase donlgerBase) {
     }
}

Reply:
Yeah so I thought about both these things - I had kept them separate since the Storage Service was specifically focused on storage methods and thought it would corrupt it to have the auth methods in there as well. And since the call to the constructor has to be the first line in the method it didn't work to create a common initialization of the auth provider. 

However, since you also feel it should just be available from the service, I'll go ahead and do that.